### PR TITLE
Assert type load behavior changes between Core and Desktop with serialization attributes

### PR DESCRIPTION
**Pull request by ViktorHofer**
Relates to https://github.com/dotnet/standard/issues/300

Had to add another project which only builds against netstandard as we are building S.R.Serialization.Formatters.Tests also against netfx to work around this bug. This is only a temporary test case and should be changed after we build against net472 which probably includes the serialization fix (see comments in code).

cc @MichalStrehovsky

### CODE REVIEW

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs [hunk 1]
```diff
@@ -102,6 +103,27 @@ private static void ValidateAndRoundtrip(object obj, string[] blobs, bool isEqua
             }
         }
 
+        [Fact]
+        public void NetStandardLibTest()
+        {
+            // When we start to run our tests against net472 we can remove the ThrowsIf clause, the NetStandardLib project
+            // and the build configuration for netfx in the Configuration.props file in the test folder. Meanwhile this asserts
```
Comment (joperezr) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs]: Do you have an issue to track this work? If so can you add it to this comment?

**Comment by joperezr** [comment]
I don't love the fact of having to add a new project just for one type in one test, is there any other serializable type in the build that we can use instead for this purpose? If not, is it possible to compile the tests against netstandard? If none of the above is possible, then the changes look good.

**Comment by ViktorHofer** [comment]
Giving it a second thought, my comment is not 100% correct. This behavior difference won't change by compiling against net472 but by running the test suite (which compiles against netstandard then) against net472. We would need a check if the runtime version is greater or equal 472. That means, I can remove the additional project.

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs [hunk 2]
```diff
@@ -102,6 +102,25 @@ private static void ValidateAndRoundtrip(object obj, string[] blobs, bool isEqua
             }
         }
 
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/standard/issues/300", TargetFrameworkMonikers.NetFramework)]
+        public void NetStandardLibTest()
+        {
+            // When net472 with the corresponding fix is released we should add a platform detection check and conditionally 
+            // run the test on .NET Core and .NET Framework >=netf472. The build configuration for netfx in the Configuration.props
+            // also needs to be removed. Meanwhile this asserts that the behavior isn't broken on .NET Cores skips on .NET Framework.
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs]: garbled comment

File: src/System.Runtime.Serialization.Formatters/tests/SerializationTypes.cs [hunk 3]
```diff
@@ -13,6 +13,38 @@ public struct TypeWithoutNamespace { }
 
 namespace System.Runtime.Serialization.Formatters.Tests
 {
+    [Serializable]
+    public class StreamingContextType
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/tests/SerializationTypes.cs]: what about OnDeserializing and OnSerialized ?

**Comment by ViktorHofer** [comment]
> Isn't this test equivalent to

Thanks, I must have overlooked that test case. Nevertheless the original purpose of this PR was to (1) assert if a StreamingContext object with a netstandard signature can be instantiated without a typeloadexception and (2) assert if netfx throws that typeloadexception.  After joperezr comment I changed that to not introduce an additional project just for verifying that the netfx behavior is broken (see commit 1-3).

I think there is no reason for this PR anymore but to add more information to the linked issue what should be done after net472 is out.

**Pull request closed without merging**